pkg/controller/clustersync/clustersync_controller.go

```diff
@@ -417,13 +430,17 @@ func (r *ReconcileClusterSync) Reconcile(ctx context.Context, request reconcile.
 		ownerRef := metav1.NewControllerRef(cd, cd.GroupVersionKind())
 		ownerRef.Controller = nil
 		clusterSync.OwnerReferences = []metav1.OwnerReference{*ownerRef}
-		if err := r.Create(context.Background(), clusterSync); err != nil {
+		switch err := r.Create(context.Background(), clusterSync); {
+		case apierrors.IsAlreadyExists(err):
+			// race condition, just proceed
```

Comment:
This would be strange, as we _shouldn't_ ever have multiple threads trying to reconcile the same cluster. I think we should at least log a warning if we get here.